maven-universe/src/main/java/org/jboss/galleon/universe/maven/repo/SimplisticMavenRepoManager.java

```diff
@@ -102,6 +103,26 @@ public void resolve(MavenArtifact artifact) throws MavenUniverseException {
         }
     }
 
+    @Override
+    public void resolveAll(List<MavenArtifact> artifacts) throws MavenUniverseException {
+        if (artifacts.stream().anyMatch(MavenArtifact::isResolved)) {
+            throw new MavenUniverseException("One of artifact is already resolved");
+        }
+
+        List<MavenArtifact> toResolve = new ArrayList<>();
+        for (MavenArtifact artifact : artifacts) {
+            Path path = getArtifactPath(artifact);
+            if (Files.exists(path)) {
+                artifact.setPath(path);
+            } else if (fallback != null) {
+                toResolve.add(artifact);
+            } else {
+                throw new MavenUniverseException(pathDoesNotExist(artifact, path));
+            }
+        }
+        fallback.resolveAll(toResolve);
```

Comment:
fallback can be null, it seems that the bug already exists in the resolve method. I am wandering if that is really needed to updated the simplistic repo manager. Is the default implementation added to the interface enough?

Reply:
Ah yes it'll break, if all the artifacts were resolved locally and fallback is null. Sorry missed that.

The default implementation would be enough for the channels. I added a full implementation so that it can be used in plain Galleon. There's a new plugin option in wildfly-galleon-plugin to control that behaviour, so if there's only the sequential default implementation in Galleon, that option would more-or-less have no effect - I thought that might be bad UX.

If you'd rather go with just the default impl, maybe I should change the plugin option for a system property?

Reply:
@spyrkob I think the change you made to https://github.com/wildfly/galleon/pull/305/files/5233f136fadb04ec23d1407c326fa0215c24d855#diff-d73733d2c1e0a6f4338e83cf723c9b14fa4fa95fb6b93c7871151bd16b1be546 is fine, we need an implementation that relies on repoSystem.resolveArtifacts for Plain Galleon. I am just wandering if overriding resolveAll in the SimplisticMavenRepoManager is required. We have now a default implementation that iterates the artifact, this should be enough for the SimplisticMavenRepoManager.

Reply:
Having a plugin option to enable the batch resolution is good with me.

Reply:
ah, I see what you mean. I'll remove this implementation, thanks!